api/pkg/backend/service.go

```diff
@@ -541,6 +602,24 @@ func (s *APIService) CreateTier(request *restful.Request, response *restful.Resp
 		return
 
 	}
+	//Validation of tenantId of tier:
+	actx := request.Attribute(c.KContext).(*c.Context)
+	token := actx.AuthToken
+	if !ValidateTenant(token, tier.TenantId) {
+		errMsg := fmt.Sprintf("tenantId:%v in request is invalid:", tier.TenantId)
+		log.Error(errMsg)
+		response.WriteError(http.StatusBadRequest, errors.New(errMsg))
+		return
+	}
+	//validation of tenants to be associated with the tier:
+	for _, tenant := range tier.Tenants {
+		if !ValidateTenant(token, tenant) {
+			errMsg := fmt.Sprintf("invalid tenant:%v is associated to tier in request:", tenant)
```

Comment:
probably msg is not clear. You may modify why it is failing. 
do you mean:??
"tenant is :%v is not valid. So it can not be associated with tier"

Reply:
modified